### ISSUE TITLE
Question: React native component update cycle and Sagas

### ISSUE THREAD

**Issue by Proberts** [yelouafi/redux-saga#656]
I'm having trouble understanding how to update a React Native UI from within the saga framework. As a simple example, I have two actions: spinnerShow() and spinnerHide() which show and hide a UI element, and a fetchData() action that, via dispatch(), triggers off fetchData_saga() using takeLatest().  If fetchData_saga() determines that it is going to take more than a few seconds I want it to call spinnerShow().   (The spinner is a Modal component that is shown/hid with a boolean state variable. fetchData_saga() has a 5-second delay() to simulate the workload.)

The Modal component UI doesn't update until the saga exits. After reading the Saga docs, what I assumed should work is using a parent saga that calls spinnerShow() then spawns fetchData_saga(), which before exiting calls spinnerHide(). However, it doesn't work - the UI doesn't update until fetchData_saga() completes. I've verified all the functions are being called and the state is being set in the middle of fetchData_saga(), and if I don't spinnerHide() the spinner appears when the Saga exits.

Google doesn't return me anything specific enough about Saga and React Native to help me understand what to do - maybe I'm googling the wrong terms, but I'm also new to React/RN and Redux. I've been at this all day and am out of ideas.

Any suggestions for resources or examples would be appreciated.

**Comment by Andarist** [comment]
Could you post your saga?

**Comment by skellock** [comment]
Hey, going to throw this hail mary out there.

Are you running `remote debugging` in React Native?  If so, turn that off for a sec.  Was chatting with a fella on Friday with the exact same problem.

**Comment by Proberts** [comment]
Thanks for the replies. I found my problem while preparing a minimal saga to post, and it was so simple I'm embarrassed to post it.

My problem was I was doing a call() to the action which just returns an event object. Switching to using put() solved my issue. It was showing the modal after the saga exited because at one point I had tried stacking two dispatch() calls, one to show the spinner, and one to my saga.
